pkg/internal/request/metric_attributes.go

```diff
@@ -118,3 +120,56 @@ func SpanPeer(span *Span) string {
 
 	return span.Peer
 }
+
+func SpanKindString(span *Span) string {
+	switch span.Type {
+	case EventTypeHTTP, EventTypeGRPC, EventTypeKafkaServer, EventTypeRedisServer:
+		return "SPAN_KIND_SERVER"
+	case EventTypeHTTPClient, EventTypeGRPCClient, EventTypeSQLClient, EventTypeRedisClient:
+		return "SPAN_KIND_CLIENT"
+	case EventTypeKafkaClient:
+		switch span.Method {
+		case MessagingPublish:
+			return "SPAN_KIND_PRODUCER"
+		case MessagingProcess:
+			return "SPAN_KIND_CONSUMER"
+		}
+	}
+	return "SPAN_KIND_INTERNAL"
+}
+
+func TraceName(span *Span) string {
```

Comment:
Since we have lots of supporting code for serialization inside `span.go` now, I wonder if this should also be moved there, analogous to my previous comment? And is there value on adding this to the debug output (like we do with the span kind)? Happy to raise a separate PR to help if that's the case.